test-interface/src/main/scala/scala/scalanative/testinterface/TestMainBase.scala

```diff
@@ -148,3 +194,29 @@ abstract class TestMainBase {
     override def trace(t: Throwable): Unit = log(Log.Level.Trace, "", Some(t))
   }
 }
+
+object TestMainBase {
+  private val faultHandler = CFunctionPtr.fromFunction1(handleSegFault _)
+  private def handleSegFault(id: Int): Unit = {
+    //making ids stable i.e. vals instead fo defs
+    //for use in the match
+    val SIGSEGV = signal.SIGSEGV
+    val SIGFPE  = signal.SIGFPE
+
+    val retCode = id match {
+      case `SIGSEGV` =>
+        Console.err.println("Segmentation fault")
+        139
+      case `SIGFPE` =>
+        Console.err.println("Erroneous Arithmetic Operation")
+        136
+      case _ =>
+        Console.err.println("Unknown fault")
+        -1
+    }
+    // the handler will run on the same thread that caused the segmentation fault
+    new Throwable().printStackTrace()
+    // segfault return code
+    System.exit(retCode)
```

Comment:
Would it be feasible to fail the current test but continue the execution after a `SIGSEGV` or `SIGFPE` (and maybe other non-fatal signals)?

Reply:
Hmm. Maybe this would work:
```scala
throw new Error(...)
//instead of
System.exit(...)
```
Need to test that.